core/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -429,18 +443,15 @@ public static ExprEval ofType(@Nullable ExpressionType type, @Nullable Object va
       case STRING:
         // not all who claim to be "STRING" are always a String, prepare ourselves...
         if (value instanceof String[]) {
-          return new ArrayExprEval(ExpressionType.STRING_ARRAY, (String[]) value);
+          return new ArrayExprEval(ExpressionType.STRING_ARRAY, Arrays.stream((String[]) value).toArray());
```

Comment:
Stream construction in hot paths can be expensive. May be better to avoid it.

Reply:
good call, looking around there are quite a few usages of steam in array stuffs, should I try to fix them all up in this PR or do as a follow-up?

Reply:
Up to you, I'm OK either way.